histogram.go

```diff
@@ -109,14 +109,36 @@ func (h *Histogram) Update(v float64) {
 	h.mu.Unlock()
 }
 
+// Merge merges histograms
+func (h *Histogram) Merge(b *Histogram) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	b.mu.RLock()
```

Comment:
does Lock cover RLock as well?

Reply:
https://pkg.go.dev/sync#RWMutex.Lock

Lock locks rw for writing. If the lock is already locked for reading or writing, Lock blocks until the lock is available. 

so just Lock is enough

Reply:
there're locks on different objects - write lock on a target histogram and read lock on src histogram

Reply:
oh sorry, makes sence ;)